Service/FederalizationiService.php

```diff
@@ -0,0 +1,283 @@
+use DateTime;
+use Doctrine\ORM\EntityManagerInterface;
+use Exception;
+use GuzzleHttp\Exception\GuzzleException;
+use Psr\Cache\CacheException;
+use Psr\Cache\InvalidArgumentException;
+use Ramsey\Uuid\Uuid;
+use Respect\Validation\Exceptions\ComponentException;
+use Symfony\Component\Console\Style\SymfonyStyle;
+use Symfony\Component\HttpFoundation\Session\SessionInterface;
+use CommonGateway\OpenCatalogiBundle\OpenCatalogiService;
+
+class FederalizationiService
+{
+    private EntityManagerInterface $entityManager;
+    private SessionInterface $session;
+    private CommonGroundService $commonGroundService;
+    private CallService $callService;
+    private SynchronizationService $synchronizationService;
+    private array $data;
+    private array $configuration;
+    private SymfonyStyle $io;
+
+    // Lets prevent unnesecery database calls
+    private Entity $catalogusEntity;
+    private Entity $componentEntity;
+    private Entity $organisationEntity;
+    private Entity $applicationEntity;
+
+
+    public function __construct(
+        EntityManagerInterface $entityManager,
+        SessionInterface $session,
+        CommonGroundService $commonGroundService,
+        CallService $callService,
+        SynchronizationService $synchronizationService
+    ) {
+        $this->entityManager = $entityManager;
+        $this->session = $session;
+        $this->commonGroundService = $commonGroundService;
+        $this->callService = $callService;
+        $this->synchronizationService = $synchronizationService;
+    }
+
+    /**
+     * Set symfony style in order to output to the console
+     *
+     * @param SymfonyStyle $io
+     * @return self
+     */
+    public function setStyle(SymfonyStyle $io):self
+    {
+        $this->io = $io;
+
+        return $this;
+    }
+
+    /**
+     * Get and handle oll the objects of an catalogi specific
+     *
+     * @param ObjectEntity $catalogus The catalogus that should be read
+     * @return array An report of the found objects
+     */
+    public function readCatalogus(ObjectEntity $catalogus):array {
+
+        // Basis setup
+        $reportOut = [];
+
+        // Check if the past object is a
+        if($catalogus->getEntity->getReferecne() != "https://opencatalogi.nl/catalogi.schema.json"){
+            (isset($this->io)?$this->io->error('The suplied Object is not of the type https://opencatalogi.nl/catalogi.schema.json'):'');
+            return $reportOut;
+        }
+
+        // Lets get the source for the catalogus
+        if(!$source = $catalogus->getValue('source')){
+            (isset($this->io)?$this->io->error('The catalogi '.$catalogus->getName.' doesn\'t have an valid source'):'');
+            return $reportOut;
+        }
+
+        (isset($this->io)?$this->io->info('Looking at '.$source->getName().'(@:'.$source->getLocation().')'):'');
+        // Lets grap ALL the objects for an external source
+        $objects = json_decode($this->callService->call(
```

Comment:
Er is een decode functie in de callserivce (misschien niet nodig hier)

Reply:
Binnen deze context verwachten we altijd json of willen we een error gooien